### ISSUE TITLE
[Console] ProgressBar behavior (question)

### ISSUE THREAD

**Issue by llaville** [symfony/symfony#13019]
Platform:  Windows with [ANSICON](https://github.com/adoxa/ansicon/releases) 1.66 installed

My Windows [Console](http://sourceforge.net/projects/console/) 2.00.148 is configured with 120 char width and 32 lines

I've migrated recently from `ProgressHelper` to `ProgressBar` with Symfony Component 2.6.1, and I've a question about the new progress bar behavior.

When full win console is reached, the progress bar (even with ANSI support) changed from a one line display to a multi lines display.

Here are what I got when the full win/console width is not yet reached. A one line display that is refreshed at each changes 

----
    3 [--->------------------------]  1 sec 7.0 MiB File Reflect\Analyser\CompatibilityAnalyser.php in progress...
----
 
And after the full win/console width is reached (in my case : 120 char) at step 52

----
    52 [------------------------>---] 13 secs 12.5 MiB File Reflect\Plugin\Cache\CacheAdapterInterface.php in progress...
    53 [------------------------->--] 13 secs 12.5 MiB File Reflect\Plugin\Cache\CacheStorageInterface.php in progress...
    54 [-------------------------->-] 13 secs 12.5 MiB File Reflect\Plugin\Cache\DefaultCacheStorage.php in progress...
    55 [--------------------------->] 13 secs 12.5 MiB File Reflect\Plugin\Cache\DoctrineCacheAdapter.php in progress...
    56 [>---------------------------] 14 secs 12.5 MiB File Reflect\Plugin\CachePlugin.php in progress...
    57 [->--------------------------] 14 secs 12.5 MiB File Reflect\Plugin\Log\DefaultLogger.php in progress...
    58 [-->-------------------------] 14 secs 12.5 MiB File Reflect\Plugin\LogPlugin.php in progress...
    59 [--->------------------------] 14 secs 12.0 MiB File Reflect\Plugin\PlantUML\PlantUMLPlugin.php in progress...
    60 [---->-----------------------] 15 secs 12.0 MiB File Reflect\Plugin\PluginInterface.php in progress...
    61 [----->----------------------] 15 secs 11.8 MiB File Reflect\Plugin\PluginManager.php in progress...
    62 [------>---------------------] 15 secs 11.8 MiB File Reflect\Plugin\ProfilerPlugin.php in progress...
    63 [------->--------------------] 16 secs 11.8 MiB File Reflect\Printer\Text.php in progress...
    64 [-------->-------------------] 16 secs 11.8 MiB File Reflect\Tokenizer\DefaultTokenizer.php in progress...
    65 [--------->------------------] 16 secs 11.8 MiB File Reflect.php in progress...
    65 [============================] 16 secs 11.8 MiB File Reflect.php in progress...
----

Is it supposed to be the standard ProgressBar behavior, or just a bug ?

**Comment by javiereguiluz** [comment]
Is anyone willing to take a look at this annoying issue before 2.7 freezes in a few days?

**Comment by sstok** [comment]
The line length seems to be messed-up because its wrapped by the console. The overwrite() method determines the length by splitting the lines, but as the line is split by the console you get an extra "\n" and thus a wrong result.

Because it does remove some chars, just not enough.
```
======================================>-
    4.78 MB/4.90 MB ====================
======================================>-
    4.80 MB/4.90 MB ====================
======================================>-
    4.81 MB/4.90 MB ====================
```

But I can't think of anything to fix this :(

**Comment by javiereguiluz** [comment]
@sstok thanks for investigating the issue. Let's see if we can find a solution somehow.

**Comment by WouterJ** [comment]
@javiereguiluz freeze is only for features, after it there are some months to fix bugs, finish docs, etc.

**Comment by llaville** [comment]
Is there any fresh news about a fix for this issue ?

**Comment by llaville** [comment]
No changes since a long time about this issue ! Should I consider to close it ?

**Comment by jakzal** [comment]
@llaville I think it's worth to try to solve it before we close.

**Comment by TomasVotruba** [comment]
Do you propose any solutions?

Should it be adjusted to the width of the window?

**Comment by xabbuh** [comment]
@TomasVotruba If possible, the progress bar should indeed limit itself to the width of the window if that is less than what would have been used otherwise.

**Comment by bamarni** [comment]
With the latest symfony installer, I'm still getting the display issue highlighted by @javiereguiluz [here](https://github.com/symfony/symfony/issues/13019#issuecomment-67627942) : 

![screenshot from 2017-09-17 17-32-00](https://user-images.githubusercontent.com/1205386/30522314-6893128a-9bce-11e7-9f26-112fda650757.png)

**Comment by ChloeSartorelli** [comment]
Not sure why this was closed? [The last time closing it was discussed](https://github.com/symfony/symfony/issues/13019#issuecomment-152007242), it sounded like the decision was to explicitly not close it without a solution:
> I think it's worth to try to solve it before we close.

I'm still seeing this problem today.

**Comment by derrabus** [comment]
It was closed because #19012 was merged as a fix.
